modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/components/rules/RuleItem.js

```diff
@@ -0,0 +1,213 @@
+	enable: Liferay.Language.get('enable'),
+	require: Liferay.Language.get('require'),
+	show: Liferay.Language.get('show'),
+};
+
+const OPERATOR_LABELS = {
+	'belongs-to': Liferay.Language.get('belongs-to'),
+	contains: Liferay.Language.get('contains'),
+	'equals-to': Liferay.Language.get('equals-to'),
+	'is-empty': Liferay.Language.get('is-empty'),
+	'not-contains': Liferay.Language.get('not-contains'),
+	'not-equals-to': Liferay.Language.get('not-equals-to'),
+	'not-is-empty': Liferay.Language.get('not-is-empty'),
+};
+
+const Text = ({capitalize = false, children = '', lowercase = false}) => (
+	<span
+		className={classNames('pr-1', {
+			'text-capitalize': capitalize,
+			'text-lowercase': lowercase,
+		})}
+	>
+		{children}
+	</span>
+);
+
+export default function RuleItem({rule, toggleRulesEditorVisibility}) {
+	const {actions, conditions, logicalOperator, name: ruleName} = rule;
+	const dispatch = useForm();
+	const {dataDefinition} = useFormState({schema: ['dataDefinition']});
+	const name = getLocalizedValue(dataDefinition.defaultLanguageId, ruleName);
+
+	const dropDownActions = [
+		{
+			action: () => toggleRulesEditorVisibility(rule),
+			name: Liferay.Language.get('edit'),
+		},
+		{
+			action: () => {
+				const confirmed = confirm(
+					Liferay.Language.get('are-you-sure-you-want-to-delete-this')
+				);
+
+				if (confirmed) {
+					dispatch({
+						payload: rule.ruleEditedIndex,
+						type: EVENT_TYPES.RULE.DELETE,
+					});
+				}
+			},
+			name: Liferay.Language.get('delete'),
+		},
+	];
+
+	const replaceExpressionLabels = (expression) => {
+		forEachDataDefinitionField(dataDefinition, ({name}) => {
+			expression = expression.replace(
+				new RegExp(`\\[${name}\\]`, 'g'),
+				getFieldLabel(dataDefinition, name)
+			);
+		});
+
+		return expression;
+	};
+
+	return (
+		<div className="rule-list__panel">
+			<ClayPanel
+				className="collapsable-panel panel-unstyled"
+				collapsable
+				collapseClassNames="panel-body"
+				displayTitle={name}
+				displayType="secondary"
+			>
+				<ClayButton
+					displayType="unstyled"
+					onClick={() => toggleRulesEditorVisibility(rule)}
+				>
+					<Text capitalize>{Liferay.Language.get('if')}</Text>
+
+					{conditions.map(({operands, operator}, index) => {
+						const [first, last] = operands;
+						const lastValue = last?.value;
+
+						const _getFieldLabel = () => {
+							const field = getDataDefinitionField(
+								dataDefinition,
+								lastValue
+							);
+
+							if (field) {
+								return getFieldLabel(dataDefinition, lastValue);
+							}
+
+							const parent = getDataDefinitionField(
+								dataDefinition,
+								first.value
+							);
+
+							if (parent) {
+								const optionLabel = getOptionLabel(
+									parent.customProperties?.options,
+									lastValue,
+									dataDefinition.defaultLanguageId
+								);
+
+								return optionLabel || lastValue;
+							}
+
+							return lastValue;
+						};
```

Comment:
There are much more sustainable ways to achieve the same information, such as go through the conditions just once and adding the corresponding field within the object, so you don't have to search all the time, instead of picking up `pages`, you would take `fields` that already do that.

This is not an improvement to do now because it is not objective and from what I understand this came from the old implementation, but I would take some lessons from the Forms RuleList that was rewritten. The implementation is simpler and easier to maintain and also avoids running through the fields nested which can degrade performance.

https://github.com/liferay/liferay-portal/blob/d96487ee50a09f95f6e87b2663b2ce03670fd275/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/js/pages/RuleList.es.js

Reply:
I agree with you, but we can resolve this at another time, and we can use the Forms RuleList as the basis for this

Reply:
Yeah, we can do that at another time is not a priority for that task.